doc/cprover-manual/history-variables.md

```diff
@@ -0,0 +1,19 @@
+# Historic variables
+
+### Denfinition
+
+`__CPROVER_old()` refers to the value of a given object
+in the pre-state of a function within the _ensures_ clause. It only supports
+scalars, structs and pointers.
+
+```c
+int add_overflow(uint32_t a, uint32_t b, uint32_t* out)
+/* ... */
+__CPROVER_ensures((__CPROVER_return_value == FAILURE) ==> (*out == __CPROVER_old(*out)))
```

Comment:
You might want to explain above what this postcondition is supposed to mean:

If the function returns a failure code, the `out` pointer should not have been modified.

Reply:
I added the description.